sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,186 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import androidx.annotation.WorkerThread
+import androidx.core.os.postDelayed
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import kotlin.math.pow
+
+@WorkerThread
+internal class FpsManager(
+  private val handler: Handler
+) {
+  private var userRefreshRate = USER_DEFINED_REFRESH_RATE_NOT_SET
+  private var userDefinedRatio = USER_DEFINED_RATIO_NOT_SET
+
+  private var screenRefreshRate = SCREEN_METRICS_NOT_DEFINED
+  private var screenRefreshPeriodNs = -1L
+  private var previousRefreshNs = -1L
+  private var previousDrawnFrameIndex = 0
+  private var frameRenderTimeAccumulatedNs = 0L
+
+  private var preRenderTimeNs = -1L
+  private var choreographerTicks = 0
+  private var choreographerSkips = 0
+
+  internal var fpsChangedListener: OnFpsChangedListener? = null
+
+  fun setScreenRefreshRate(screenRefreshRate: Int) {
+    if (this.screenRefreshRate == screenRefreshRate) {
+      return
```

Comment:
nit:
nice to add boolean flag as a result or throw some exception

Reply:
why? if we did get here - it means that device notified again that screen refresh rate has arrived (which happens e.g. if we minimize / maximize app). It's valid situation that we simply return.